data/src/main/java/org/apache/iceberg/data/GenericRecord.java

```diff
@@ -104,7 +100,7 @@ public Object getField(String name) {
   @Override
   public void setField(String name, Object value) {
     Integer pos = nameToPos.get(name);
-    Preconditions.checkArgument(pos != null, "Cannot set unknown field named: " + name);
+    Preconditions.checkNotNull(pos, "Cannot set unknown field named: %s", name);
```

Comment:
Actually, I think this should be `checkArgument`. The argument is invalid because `pos` is null, but the problem is that the name was not found, not that the name was null. So `NullPointerException` would be misleading because `name` may not be null to produce it.

Reply:
+1. I'll revert